app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -401,52 +398,50 @@ class AcraWrapper {
     }
 }
 
-class UsageUtility(private val prefs: SharedPreferences) {
+class UserFeedbackUtility(private val prefs: SharedPreferences) {
     private val numberOfTimesOpenedKey = "numberOfTimesOpened"
     private val userGaveFeedbackKey = "userGaveFeedback"
     private val dateTimeFirstOpenKey = "dateTimeFirstOpen"
     private val millisecondsInThreeDays = 259200000L
     private val minimumNumberOfOpensBeforeReviewRequest = 15
 
-    fun addToNumberOfTimesOpened(): Int {
-        val numberTimesOpened = prefs.getInt(numberOfTimesOpenedKey, 1)
-
-        if (numberTimesOpened == 1) {
-            with(prefs.edit()) {
-                putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
-                putLong(dateTimeFirstOpenKey, Date().time)
-                apply()
-            }
-        } else {
-            with(prefs.edit()) {
-                putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
-                apply()
-            }
-        }
+    fun askingForFeedbackIsAppropriate(): Boolean {
+        val minimumTimePassedReached = getIsSufficientTimeElapsedSinceFirstOpen()
+        val minimumTimesOpenedReached = numberOfTimesOpenedIsGreaterThanThreshold()
+        val userAlreadyGaveReviewOrDismissed = getUserGaveFeedback()
 
-        return numberTimesOpened
+        return minimumTimePassedReached && minimumTimesOpenedReached && !userAlreadyGaveReviewOrDismissed
```

Comment:
This might be handled by the compiler, but you could save a bit of work by calling the functions inline in the return. That way, as soon as one fails, the other functions won't be even be called.

Reply:
I've changed the function so it calls all three sub-functions inline